src/key.rs

```diff
@@ -589,6 +627,25 @@ impl PublicKey {
             }
         }
     }
+
+    /// Returns the [`XOnlyPublicKey`] (and it's [`Parity`]) for this [`PublicKey`].
+    #[inline]
+    pub fn x_only_public_key<C: Signing>(&self, secp: &Secp256k1<C>) -> Result<(XOnlyPublicKey, Parity), Error> {
```

Comment:
The call to `Parity::from_i32` can fail. Or is this failure not actually possible, does libsecp256k1 always return a valid parity so we can use `expect`?

Reply:
Yes, as per documentation:
```
 *          pk_parity: pointer to an integer that will be set to 1 if the point
 *                     encoded by xonly_pubkey is the negation of the pubkey and
 *                     set to 0 otherwise. (can be NULL)
```

Reply:
See: https://github.com/bitcoin-core/secp256k1/blob/8746600eec5e7fcd35dabd480839a3a4bdfee87b/include/secp256k1_extrakeys.h#L88

Reply:
Thanks man, I would have also accepted the answer 'RTFM Tobin!' :)